### PR TITLE
Feature: Form Basics Lesson

### PR DESCRIPTION
**Pull request by KevinMulhern**
Closes: https://github.com/TheOdinProject/top-meta/issues/25

Because:
* We need a lesson to teach students about HTML forms.

This commit:
* Covers the form element and its attributes
* Covers all the common form controls
* Covers how to guide users to enter the correct information using labels and placeholders
* Covers the name attribute using an example by submitting to httpbin to see the data the server will receive when a form is submitted
* Covers organizing forms with fieldsets
* Touches briefly on styling in the lesson content and provides some assignments to go deeper. I wanted to have a full lesson on styling but it wasn't going to be of the same quality as the linked resources

 - [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [x] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
 - [x] You have previewed your markdown (if any) in the [Odin Markdown Preview Tool](https://www.theodinproject.com/lessons/preview)
 - [ ] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1. Describe the changes made and include why they are necessary or important:

...your text here

#### 2. Related Issue

Closes #XXXXX

### CODE REVIEW

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+<label for="firstName">First Name:</label>
+<input type="text" id="first_name" name="first_name">
+~~~
+
+The name attribute serves as a reference to the data inputted into a form control after submitting it. You can think of it as a variable name for the input. Form input should always have a name attribute; otherwise, it will be ignored when the form is submitted.
+
+To get a better understanding of what this looks like we can submit a form to [httpbin](http://httpbin.org/). This service will send back a response which will let us view what data was submitted. Fill in the form below and click submit:
+
+<p class="codepen" data-height="300" data-theme-id="dark" data-default-tab="html,result" data-slug-hash="dyVRMbq" data-preview="true" data-user="TheOdinProjectExamples" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
+<span>See the Pen <a href="[https://codepen.io/TheOdinProjectExamples/pen/dyVRMbq](https://codepen.io/TheOdinProjectExamples/pen/dyVRMbq)">
```
Comment (zachmmeyer) [inline html_css/v2/main-course/intermediate_html/form-basics.md:102]:
```suggestion
<span>See the Pen <a href="https://codepen.io/TheOdinProjectExamples/pen/dyVRMbq">
```

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+Try changing the name attributes of some of the input fields in the form and then submitting again to see how form data in the response changes.
+
+**Using Form Controls Outside of Forms**
+
+It's worth mentioning that you can use any of the form controls HTML provides outside of the `<form>` element, even when you don't have a backend server where you can send data.
+
+For example you might want to have an input that gets some data from a user and display that somewhere else on the page with Javascript:
+
+<p class="codepen" data-height="300" data-theme-id="dark" data-default-tab="html,result" data-slug-hash="PoJjNYr" data-preview="true" data-user="TheOdinProjectExamples" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
+<span>See the Pen <a href="[https://codepen.io/TheOdinProjectExamples/pen/PoJjNYr](https://codepen.io/TheOdinProjectExamples/pen/PoJjNYr)">
```
Comment (zachmmeyer) [inline html_css/v2/main-course/intermediate_html/form-basics.md:127]:
```suggestion
<span>See the Pen <a href="https://codepen.io/TheOdinProjectExamples/pen/PoJjNYr">
```

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+~~~
+
+The name attribute serves as a reference to the data inputted into a form control after submitting it. You can think of it as a variable name for the input. Form input should always have a name attribute; otherwise, it will be ignored when the form is submitted.
+
+To get a better understanding of what this looks like we can submit a form to [httpbin](http://httpbin.org/). This service will send back a response which will let us view what data was submitted. Fill in the form below and click submit:
+
+<p class="codepen" data-height="300" data-theme-id="dark" data-default-tab="html,result" data-slug-hash="dyVRMbq" data-preview="true" data-user="TheOdinProjectExamples" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
+<span>See the Pen <a href="[https://codepen.io/TheOdinProjectExamples/pen/dyVRMbq](https://codepen.io/TheOdinProjectExamples/pen/dyVRMbq)">
+form-basics-name-attribute</a> by TheOdinProject (<a href="[https://codepen.io/TheOdinProjectExamples](https://codepen.io/TheOdinProjectExamples)">@TheOdinProjectExamples</a>)
```
Comment (zachmmeyer) [inline html_css/v2/main-course/intermediate_html/form-basics.md:103]:
```suggestion
form-basics-name-attribute</a> by TheOdinProject (<a href="https://codepen.io/TheOdinProjectExamples">@TheOdinProjectExamples</a>)
```

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+on <a href="[https://codepen.io](https://codepen.io/)">CodePen</a>.</span>
```
Comment (zachmmeyer) [inline html_css/v2/main-course/intermediate_html/form-basics.md:104]:
```suggestion
on <a href="https://codepen.io/">CodePen</a>.</span>
```

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+
+**Using Form Controls Outside of Forms**
+
+It's worth mentioning that you can use any of the form controls HTML provides outside of the `<form>` element, even when you don't have a backend server where you can send data.
+
+For example you might want to have an input that gets some data from a user and display that somewhere else on the page with Javascript:
+
+<p class="codepen" data-height="300" data-theme-id="dark" data-default-tab="html,result" data-slug-hash="PoJjNYr" data-preview="true" data-user="TheOdinProjectExamples" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
+<span>See the Pen <a href="[https://codepen.io/TheOdinProjectExamples/pen/PoJjNYr](https://codepen.io/TheOdinProjectExamples/pen/PoJjNYr)">
+using-form-controls-outside-of-form</a> by TheOdinProject (<a href="[https://codepen.io/TheOdinProjectExamples](https://codepen.io/TheOdinProjectExamples)">@TheOdinProjectExamples</a>)
```
Comment (zachmmeyer) [inline html_css/v2/main-course/intermediate_html/form-basics.md:128]:
```suggestion
using-form-controls-outside-of-form</a> by TheOdinProject (<a href="https://codepen.io/TheOdinProjectExamples">@TheOdinProjectExamples</a>)
```

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+on <a href="[https://codepen.io](https://codepen.io/)">CodePen</a>.</span>
```
Comment (zachmmeyer) [inline html_css/v2/main-course/intermediate_html/form-basics.md:129]:
```suggestion
on <a href="https://codepen.io">CodePen</a>.</span>
```

**Review by zachmmeyer — COMMENTED** [review]
This lesson is great!

I had some bugs with the codepen embeds. It looks like there's markdown mixed with html tags for the links, making them broken for me. I suggested fixes, but I have no idea if it was done for a reason. Same deal with the scripts below the embeds. I didn't suggest those be changed because modifying them didn't actually change anything.

**Comment by KevinMulhern** [comment]
Thanks Zach, great spot on the embed links! That's from copy and pasting from notion 🙈

**Review by leosoaivan — COMMENTED** [review]
LGTM, left a minor comment re: grammar.

File: html_css/v2/main-course/intermediate_html/form-basics.md [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+
+- Create forms with HTML
+- Have a basic idea of how to style forms
+
+### The Form Element
+
+The form element is a container element like the div element we learned earlier in the curriculum. The form element wraps all of the inputs a user will interact with on a form.
+
+The form element accepts two essential attributes; the first is the `action` attribute which takes a URL value that tells the form where it should send its data to be processed.
+Later in the curriculum, we will learn to hook backend systems up to front-end forms using this attribute. For now, it's only essential to know what the action attribute is used for.
```
[the remark this answers is not in the document]
Comment (KevinMulhern) [reply html_css/v2/main-course/intermediate_html/form-basics.md]: Very good point, I think we ended up going with the non-hyphenated version https://github.com/TheOdinProject/curriculum/pull/22354

Updated, thanks @leosoaivan

**Review by zachmmeyer — APPROVED** [review]

**Pull request merged**